tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -42,128 +28,77 @@ public class Tracer : IDatadogTracer
-            if (!string.IsNullOrWhiteSpace(Settings.CustomSamplingRules))
-            {
-                foreach (var rule in CustomSamplingRule.BuildFromConfigurationString(Settings.CustomSamplingRules))
-                {
-                    Sampler.RegisterRule(rule);
-                }
-            }
-
-            if (Settings.GlobalSamplingRate != null)
-            {
-                var globalRate = (float)Settings.GlobalSamplingRate;
-
-                if (globalRate < 0f || globalRate > 1f)
-                {
-                    Log.Warning("{ConfigurationKey} configuration of {ConfigurationValue} is out of range", ConfigurationKeys.GlobalSamplingRate, Settings.GlobalSamplingRate);
-                }
-                else
-                {
-                    Sampler.RegisterRule(new GlobalSamplingRule(globalRate));
-                }
-            }
-
-            // Register callbacks to make sure we flush the traces before exiting
-            LifetimeManager.Instance.AddShutdownTask(RunShutdownTasks);
-
-            // start the heartbeat loop
-            _heartbeatTimer = new Timer(HeartbeatCallback, state: null, dueTime: TimeSpan.Zero, period: TimeSpan.FromMinutes(1));
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Tracer"/> class.
+        /// For testing only.
+        /// Note that this API does NOT replace the global Tracer instance.
+        /// The <see cref="TracerManager"/> created will be scoped specifically to this instance.
+        /// </summary>
+        internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler sampler, IScopeManager scopeManager, IDogStatsd statsd)
+            : this(TracerManagerFactory.Instance.CreateTracerManager(settings, agentWriter, sampler, scopeManager, statsd, runtimeMetrics: null, libLogSubscriber: null))
+        {
+        }
 
-            // If configured, add/remove the correlation identifiers into the
-            // LibLog logging context when a scope is activated/closed
-            if (Settings.LogsInjectionEnabled)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Tracer"/> class.
+        /// Should only be called DIRECTLY for testing purposes.
+        /// If non-null the provided <see cref="TracerManager"/> will be tied to this TracerInstance (for testing purposes only)
+        /// If null, the global <see cref="TracerManager"/> will be fetched or created, but will not be modified.
+        /// </summary>
+        private protected Tracer(TracerManager tracerManager)
+        {
+            _tracerManager = tracerManager;
+            if (tracerManager is null)
             {
-                InitializeLibLogScopeEventSubscriber(_scopeManager, DefaultServiceName, Settings.ServiceVersion, Settings.Environment);
+                // Ensure the global TracerManager instance has been created
+                // to kick start background processes etc
+                _ = TracerManager.Instance;
```

Comment:
Should this be `_tracerManager = TracerManager.Instance;` ?

Reply:
No - setting `_tracerManager` here would tie this instance of the `Tracer` to a specific `TracerManager` instance (the current global one). Tying a `Tracer` instance to a `TracerManager` is something that should only happen in tests, not in "normal" usage. Otherwise, calling `Tracer.ReplaceGlobalSettings()` would not seemlessly swap `TracerManager` for all `Tracer` instances

It's unfortunate that this constructor needs to be called at all in "normal usage", but our hands are tied somewhat by the existing constructors